code/datums/datum.dm

```diff
@@ -28,7 +28,7 @@
 	active_timers = null
 	for(var/thing in timers)
 		var/datum/timedevent/timer = thing
-		if (timer.spent)
+		if (timer.spent && !(timer.flags & TIMER_DELETE_ME))
```

Comment:
whats with this flag? are we going to need to add that in everywhere now to ensure timers get deleted properly? Why would that not be default behaviour

Reply:
Don't worry about that. It's used to qdelete TIMER_LOOP

Reply:
i don't quite follow, nothing about loops is being checked here.

It seems to govern whether timers are deleted when the parent datum is. I cant think of any situations we wouldn't want to do that, 
how is it limited to loops?

Reply:
removed that flag